tools/meson.build

```diff
@@ -25,11 +25,14 @@ libarchive = dependency('libarchive', required: false)
 if not libarchive.found()
     # homebrew
     libarchive_lib = cc.find_library('archive', dirs: ['/usr/local/opt/libarchive/lib'], required: false)
-    if not libarchive_lib.found()
```

Comment:
The problem is just include_directories, right? Instead of this, let's make a variable similar to libarchive_lib for the different include directories. Something like

```
libarchive_lib = ...
libarchive_includes = ['/usr/local/opt/libarchive/include']
if not libarchive_lib.found()
	libarchive_lib = ...
	libarchive_includes = ['/opt/homebrew/opt/libarchive/lib']
endif
```

Reply:
Not quite, I think?  Depends on what you mean.  The problem was that whether your Mac has Homebrew in `/usr/local/opt` (x86) or `/opt` (Apple Silicon), it was always choosing `/usr/local/opt/libarchive/include`.  That's obviously inconsistent on Apple Silicon, and it doesn't work on my Apple Silicon machine.  Maybe your machine has a compatibilty symlink for that and mine doesn't?

Also, the existing code allows builds (without fakefsify) when libarchive is not present, I think your proposed change would break that.

I left this change as-is for now but I'll revise if you still want.